src/uct/tcp/tcp_md.c

```diff
@@ -47,25 +58,61 @@ static ucs_status_t uct_tcp_mem_dereg(uct_md_h uct_md,
     return UCS_OK;
 }
 
+static void uct_tcp_md_close(uct_md_h md)
+{
+    uct_tcp_md_t *tcp_md = ucs_derived_of(md, uct_tcp_md_t);
+    ucs_free(tcp_md);
+}
+
+static uct_md_ops_t uct_tcp_md_ops = {
+    .close              = uct_tcp_md_close,
+    .query              = uct_tcp_md_query,
+    .mkey_pack          = ucs_empty_function_return_success,
+    .mem_reg            = uct_tcp_md_mem_reg,
+    .mem_dereg          = uct_tcp_mem_dereg,
+    .detect_memory_type = ucs_empty_function_return_unsupported
+};
+
 static ucs_status_t
 uct_tcp_md_open(uct_component_t *component, const char *md_name,
-                const uct_md_config_t *md_config, uct_md_h *md_p)
+                const uct_md_config_t *uct_md_config, uct_md_h *md_p)
 {
-    static uct_md_ops_t md_ops = {
-        .close              = ucs_empty_function,
-        .query              = uct_tcp_md_query,
-        .mkey_pack          = ucs_empty_function_return_success,
-        .mem_reg            = uct_tcp_md_mem_reg,
-        .mem_dereg          = uct_tcp_mem_dereg,
-        .detect_memory_type = ucs_empty_function_return_unsupported
-    };
-    static uct_md_t md = {
-        .ops          = &md_ops,
-        .component    = &uct_tcp_component
-    };
-
-    *md_p = &md;
+    const uct_tcp_md_config_t *md_config = ucs_derived_of(uct_md_config,
+                                                          uct_tcp_md_config_t);
+    uct_tcp_md_t *tcp_md;
+    ucs_status_t status;
+    int i;
+
+    tcp_md = ucs_malloc(sizeof(uct_tcp_md_t), "uct_tcp_md_t");
+    if (NULL == tcp_md) {
+        ucs_error("failed to allocate memory for uct_tcp_md_t");
+        status = UCS_ERR_NO_MEMORY;
+        goto err;
+    }
```

Comment:
btw, why we need this?
actually, TCP has a single domain.
so, it could be:
```
    static uct_tcp_md_t md = {
        .super = {
            .ops       = &md_ops,
            .component = &uct_tcp_component
        }
    };

    md.config.af_prio_count= ucs_max(md_config->af_prio_count, 1);

    for (...
    ...

    *md_p = (uct_md_h)&md;
```

Reply:
I saw that this is the way done for other MDs if there is any additional config

Reply:
yes, if we have config cannot have a single global md